HW05.Rmd

```diff
@@ -0,0 +1,159 @@
+---
+title: "HW05"
+author: "Xing Yanfeng"
+date: "2020/8/2"
+output: github_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## Assignment 1
+
+### Load libraries and read in data.
+
+```{r libraries data}
+# load libraries
+library(tidyr)
+library(dplyr)
+library(readr)
+library(ggplot2)
+# read in data
+dad_mom <- read_delim("dad_mom.txt", delim = "\t")
+```
+
+### Tidying data
+
+```{r tidying}
+tidy_dad_mom <- dad_mom %>% 
+  gather(key = "key", value = "value", 2:5) %>% 
+  separate(col = key, into = c("key", "relationship")) %>% 
+  spread(key = key, value = value, convert = T) %>% 
+# rearrange columns to make it more comfortable
+  select(fam_id, relationship, name, income)
+print(tidy_dad_mom)
+```
+
+## Assignment 2
+
+### Read in data
+
+```{r read in data}
+AUC_1 <- read_csv("CTRP_files/AUC_1.csv")
+AUC_2 <- read_csv("CTRP_files/AUC_2.csv")
+cell_line <- read_csv("CTRP_files/cancer_cell_line_info.csv")
+compound <- read_csv("CTRP_files/compound_info.csv")
+experiment <-  read_csv("CTRP_files/Experiment_info.csv")
+```
+
+### Join 5 csv files
+
+```{r join 5 csv files}
+cancer_auc <- AUC_1 %>%
+# bind AUC_1 and AUC_2
+  bind_rows(AUC_2)  %>%
+# add experiment information
+  left_join(experiment, by = c("experiment_id" = "expt_id"))  %>%
+# add compound information
+  left_join(compound, by = "master_cpd_id") %>% 
+# add cell line information
+  left_join(cell_line, by = "master_ccl_id")
+```
+
+### Question 1
+
+Which cancer type has the lowest AUC values to the compound "vorinostat"?
+
+```{r question 1 data}
+vorinostat <- cancer_auc %>% 
+# filter AUC values and cancer type of vorinostat treated experiments
+  select(area_under_curve, cpd_name, cancer_type) %>% 
+  filter(cpd_name == "vorinostat") %>% 
+  drop_na(cancer_type) %>% 
+# calculate average AUC values
+  group_by(cancer_type) %>% 
+  summarize(AUC = round(mean(area_under_curve), digits = 2)) %>% 
+  arrange(AUC) %>% 
+  mutate(cancer_type = factor(cancer_type, levels = cancer_type))
```

Comment:
A clean and straightforward way to select and arrange the data that you're interested in, especially in the second part (summarize -> arrange -> mutate). I decided to use boxplot instead and even though it still allowed me to answer the question correctly, the delivery wasn't as clean as in your case.